Bowling.java

```diff
@@ -138,19 +149,20 @@ public Integer getScore() {
 	}
 
 	public Boolean isGameOver() {		
-		return allFrames.size() == 11;
+		return isFinalFrame() && currentFrame().isFull();
 	}
 	
 	public Boolean isFinalFrame() {
-		return allFrames.size() >= 10;
+		return currentFrame() instanceof FinalFrame;
```

Comment:
Interesting that we could still be checking that `allFrames.size() == 10` instead of that the class is FinalFrame.  I'm not sure there's a huge benefit either way, just noting it.